src/modules/commander/Commander.cpp

```diff
@@ -497,6 +497,134 @@ transition_result_t Commander::disarm(arm_disarm_reason_t calling_reason)
 	return arming_res;
 }
 
+transition_result_t
+Commander::try_mode_change(main_state_t desired_mode, const bool enable_fallback, const bool notify_user)
```

Comment:
I would prefer the boolean flags to be either removed or replaced with `enum class` to have type safety. Otherwise, it's easy to get it wrong when calling the method.

Reply:
We can also change that in a later step.

Reply:
given how long this PR is dragging i would leave it as it is for now, i can't allocate further time to this

Reply:
added an enum :+1: